### DISCUSSION TITLE
Looking at contributing having trouble building.

### DISCUSSION THREAD

**Discussion by jl-rey** [dotnet/docfx#7026] [Q&A]
Hello:

I have cloned this and downloaded to my computer.

Loaded into Visual Studio 2019. Cannot get to build.

I found two help wanted that looked fairly isolated and simple.

### Build with Visual Studio

Tried to build inside Visual Studio.
  First build had 687 errors.
  Second attempt 300+ errors.
  Third attempt 17 errors.

It would seem that some of the project build dependencies are not set properly

Now I am down to errors about not being able to find:

- ManagedREference.FSharp.dll
- MangedReference.dll

#### FSharpProjects.code-workspace
I found an FSharpProrojects.code-workspace.

#### Questions

How do I get these things to link?
Why does the F# code not appear in docfx.sln project.
Why is it not built with the solution.

### Visual Studio code

In the directory containing docfx.sln opened a terminal

Get the following

```
Running on OS Windows
ERROR: UpdateTemplate.sh requires node installed globally.
Error 1: Update template
```
This seems to fail on this _UpdateTempleate.ps1_

``` PowerShell
# Check if node exists globally
if (-not (Get-Command "node" -ErrorAction SilentlyContinue)) {
    Write-Host "ERROR: UpdateTemplate.sh requires node installed globally."
    exit 1
}
```
#### Questions

Exactly what do I need to install globally?

**Comment by jl-rey** [comment]
I would like to contribute but am having difficulty getting this to compile.

Is there a set of instructions for building this solution?

**Comment by minecraftchest1** [comment]
You need to install node.js globally so that the build scripts have access to the node.js commands.
